### PR TITLE
faac: refactor and add drm optional

### PR DESCRIPTION
**Pull request by codyopel**

### CODE REVIEW

File: pkgs/development/libraries/faac/default.nix [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
 
-  meta = {
+  meta = with stdenv.lib; {
     description = "Open source MPEG-4 and MPEG-2 AAC encoder";
-    homepage = http://www.audiocoding.com/faac.html;
-    # Incompatible with GPL. Some changes to the base code, included in faac,
-    # are under LGPL though.
-    license = stdenv.lib.licenses.unfreeRedistributable;
+    homepage    = http://www.audiocoding.com/faac.html;
+    license     = stdenv.lib.licenses.unfree;
```
Comment (wmertens) [inline pkgs/development/libraries/faac/default.nix]: @codyopel why did you change the license from unfreeRedistributable?

Comment (codyopel) [inline pkgs/development/libraries/faac/default.nix]: According to faac, Dolby holds patent licenses making it a patent violation to redistribute binaries.

Comment (codyopel) [inline pkgs/development/libraries/faac/default.nix]: @wmertens The best I can tell is that regardless of patents, other distros do build faac as non-free software, so I will revert my licensing change.  Thanks for bringing it to my attention.
